misc/bstr.c

```diff
@@ -100,7 +100,8 @@ int bstrspn(struct bstr str, const char *accept)
 
 int bstr_find(struct bstr haystack, struct bstr needle)
 {
-    for (int i = 0; i < haystack.len; i++)
+    size_t n = haystack.len > needle.len ? haystack.len - needle.len + 1 : 0;
+    for (int i = 0; i < n; i++)
         if (bstr_startswith(bstr_splice(haystack, i, haystack.len), needle))
```

Comment:
`bstr_startswith(struct bstr str, struct bstr prefix)` will always return `false` if `str.len < prefix.len`
When length of `bstr_splice(haystack, i, haystack.len)` is less than `needle`, this function should return already